clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -149,9 +149,21 @@ invariant_checker_t::parallel_shard_init_stream(int shard_index, void *worker_da
     auto per_shard = std::unique_ptr<per_shard_t>(new per_shard_t);
     per_shard->stream = shard_stream;
     void *res = reinterpret_cast<void *>(per_shard.get());
-    std::lock_guard<std::mutex> guard(shard_map_mutex_);
+    std::lock_guard<std::mutex> guard(init_mutex_);
     per_shard->tid_ = shard_stream->get_tid();
     shard_map_[shard_index] = std::move(per_shard);
+    // If we are dealing with an OFFLINE_FILE_TYPE_ARCH_REGDEPS trace, we need to set the
+    // dcontext ISA mode to the correct synthetic ISA (i.e., DR_ISA_REGDEPS).
+    uint64_t filetype = shard_stream->get_filetype();
+    if (TESTANY(OFFLINE_FILE_TYPE_ARCH_REGDEPS, filetype)) {
+        dr_isa_mode_t isa_mode = dr_get_isa_mode(drcontext_);
+        if (isa_mode != DR_ISA_REGDEPS) {
+            dr_set_isa_mode(drcontext_, DR_ISA_REGDEPS, nullptr);
+            std::cerr << "WARNING: invariant_checker is begin run on an "
```

Comment:
We're single-threaded here, so we only print once, although I wish there was a better place to print this.

Reply:
You mean you want to avoid printing again on each shard, so thousands of prints for a many-sharded trace?  There's no global control point today that has the filetype.  We could easily print once at the end during global aggregation: nicer up front but doesn't seem so bad at the end as the last line?

Reply:
Done.